book/core_programming.md

```diff
@@ -2,153 +2,310 @@
 
 The principles outlined in this chapter represent good practices for general programming and software development.
 
-Before reading this chapter, you would benefit from having an understanding of core programming concepts.
-This includes assigning and using variables, and using functions and classes.
+However, they are tailored to a more analytical workflow and would benefit analysts with understanding of core programming concepts such as variables, functions and classes. You can find more information and helpful resources in the [resources](resources) section.
 
+## Motivation
 
-```{admonition} Pre-requisites
```

Comment:
I think some discussion of pre-requisites up front is useful. We should retain this but make sure that it just lists pre-requisite skills rather than resources here.

Reply:
I'll collate all of the training links on a dedicated resources page